iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -619,6 +619,7 @@ public async Task DisableMethodsAsync(TimeSpan timeout)
             finally
             {
                 Logging.Exit(this, timeout, nameof(DisableMethodsAsync));
+                _methodLinkSemaphore.Release();
```

Comment:
You'll want to expand the view to see the whole method here. At line 593, this semaphore is grabbed. After that, the try block does some logic that may or may not throw an exception, so this finally block is the perfect place to release the semaphore